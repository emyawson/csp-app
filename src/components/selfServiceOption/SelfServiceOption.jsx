/* eslint-disable react/prop-types */
import React, { Fragment } from 'react';
import Button from '@ingka/button';
import RecommendedLabel from '../recommendedLabel/RecommendedLabel.jsx';

// Button
import '@ingka/svg-icon/style.scss';
import '@ingka/button/style.scss';
import '@ingka/focus/style.scss';
import RichTextEditor from '../RichTextEditor/RichText.jsx';
import './SelfServiceOption.scss';
import StillNeedHelp from '../stillNeedHelp/StillNeedHelp.jsx';

const SelfServiceOption = ({
  headline, content, buttonLink, buttonLabel, imgCMS,
}) => {
  const cardClass = () => (!imgCMS ? 'selfservice' : 'selfservice-with-image' + ' selfservice');
  return (
    <Fragment>
      <div className={cardClass()}>
        {imgCMS && (
          <div
            className="selfservice-with-image__picture"
            data-cytest="selfservice-image"
          >
            <img src={imgCMS} role="presentation" />
          </div>
        )}
        <div>
          <div className="selfservice__label" data-cytest="recommended-label">
            <RecommendedLabel />
          </div>
          {/* Description and title DESKTOP */}
          <h2 className="selfservice__title" data-cytest="selfservice-headline">
            {headline}
          </h2>
          <div className="selfservice__description" data-cytest="selfservice-description">
            <RichTextEditor data={content} />
          </div>

          {buttonLabel !== null && (
            <Button
              data-cytest="button-selfService"
              onClick={() => {
                window.location.href = buttonLink;
              }}
              className="selfservice__button"
              type="primary"
            >
              {buttonLabel}
            </Button>
          )}
        </div>
      </div>
      <div className="separator" />
      {/* This component shows as a way to present the contact cards. */}
      <StillNeedHelp />
    </Fragment>
  );
};

export default SelfServiceOption;
