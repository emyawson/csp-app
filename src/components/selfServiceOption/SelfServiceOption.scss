@import '../../index';

.selfservice__button {
  margin-top: 24px;
}
.selfservice__title {
  @extend %large-titlebase;
  margin-bottom: 16px;
}
.selfservice__description {
  @extend %description-copybase;
  @include respond(tablet) {
    max-width: 467px;
  }
  @include respond(desktop) {
    max-width: 504px
  }
}
.selfservice-with-image {
  display: grid;
  column-gap: 42px;
  grid-template-columns: 1fr;
  @include respond(desktop) {
    max-width: 1239px;
    flex-direction: row;
    grid-template-columns: 1fr 1fr;
  }
}
.selfservice-with-image__picture {
  @include respond(desktop) {
    order: 1;
  }
}
.selfservice-with-image__picture img {
  width:100%;
  height: auto;
  margin-top: 24px;
  min-width: 335px;
  @include respond(tablet) {
    max-width: 469px;
  }
  @include respond(desktop) {
    max-width: 100%
  }
}