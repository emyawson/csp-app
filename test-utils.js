import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
