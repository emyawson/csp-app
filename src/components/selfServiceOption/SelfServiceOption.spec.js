import React from 'react';
// import { render, screen } from "@testing-library/react";
import SelfServiceOption from './SelfServiceOption.jsx';
// import { renderWithRouter } from "../test-utils";
import { renderWithRouter } from '../../../test-utils';

const defaultState = {};

const mockProps = {

  selfServiceHeadline: 'Headline self service',
  selfServiceButton: 'label button',
  selfServiceContent: {
    content: [
      { value: 'need body text' },
    ],
  },
  selfServiceButtonExists: true,
  img: 'asset/test.jpg',

};

const renderComponent = (state = {}, props) => {
  const combinedState = {
    ...defaultState,
    ...state,
  };
  return renderWithRouter(<SelfServiceOption {...mockProps} {...props} />, combinedState);
};

test('renders <SelfServiceOption/>', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();

  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
