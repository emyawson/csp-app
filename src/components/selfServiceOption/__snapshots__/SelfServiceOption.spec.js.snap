// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <SelfServiceOption/> 1`] = `
<div>
  <div
    class="selfservice"
  >
    <div>
      <div
        class="selfservice__label"
        data-cytest="recommended-label"
      >
        <p
          class="label"
        >
          Aanbevolen
        </p>
      </div>
      <h2
        class="selfservice__title"
        data-cytest="selfservice-headline"
      />
      <div
        class="selfservice__description"
        data-cytest="selfservice-description"
      >
        <div
          class="richtext-editorstyle "
        />
      </div>
      <button
        class="btn btn--primary selfservice__button"
        data-cytest="button-selfService"
        type="button"
      >
        <span
          class="btn__inner"
        >
          
        </span>
      </button>
    </div>
  </div>
  <div
    class="separator"
  />
  <div
    class="section"
  >
    <h2
      class="section__title"
    >
      Nog hulp nodig?
    </h2>
    <div
      class="section__body"
    >
      Heb je een andere vraag of een probleem  over de beschikbaarheid van producten en vind je het antwoord niet? Neem contact op met de klantenservice, zij helpen je verder.
    </div>
  </div>
</div>
`;
